### PR TITLE
Add functions to evaluate models on SDSS

### LINKED ISSUE #283: move code from `case_studies/sdss_galaxies` to bliss codebase
- [ ] looping over chunks and getting/saving variational parameters should be done in `predict.py`
- [ ]  confusion matrix and precision/recall could be used by other parts of the code, so maybe useful to put in a file called `reporting.py`

### PR DESCRIPTION
**Pull request by ismael-mendoza**
Closes #283 

Next PR will simplify code for reconstructions, search for new blends and make it easy to plot them.

- [x] Functions in `predict.py` to get variational parameters on all tiles on all scene and map estimates on full scene
- [x] functions on `report.py` to load results from predict and report metrics `precision, recall, f1, classification accuracy, confusion matrix`
- [x] (maybe) functions on `report.py` to create plots of metrics as a function of magnitude and example reconstructions

### CODE REVIEW

File: bliss/predict.py [hunk 1]
```diff
@@ -27,107 +52,190 @@ def prediction(image, image_encoder, galaxy_encoder=None, binary_encoder=None):
...
+                    # locations in pixels consistent with full scene.
+                    # 0.5 comes from pt/pr definition (plotting both -> off by half a pixel).
+                    x = est_locs[:, 1].reshape(-1, 1) * clen + x1 - 0.5
+                    y = est_locs[:, 0].reshape(-1, 1) * clen + y1 - 0.5
+                    est_locs = torch.hstack((x, y)).reshape(-1, 2)
+
+                    # get fluxes from galaxy parameters by passing them though galaxy_decoder.
+                    latents = est_params["galaxy_params"].to(device).reshape(-1, latent_dim)
+                    single_galaxies = galaxy_decoder(latents)
+                    est_galaxy_flux = single_galaxies.sum((-1, -2, -3)).cpu().reshape(-1)
```
Comment (ismael-mendoza) [inline bliss/predict.py:173]: slack on bins of magnitude

File: bliss/predict.py [hunk 2]
```diff
@@ -5,19 +5,44 @@
...
 from bliss.sleep import SleepPhase
 
 
-# TODO: Go back here: https://gist.github.com/ismael-mendoza/85e6c245a177c82b071557d9d097ebc0
-def prediction(image, image_encoder, galaxy_encoder=None, binary_encoder=None):
+def predict_on_image(
+    image: torch.Tensor,
+    image_encoder: encoder.ImageEncoder,
+    binary_encoder: BinaryEncoder,
+    galaxy_encoder: GalaxyEncoder,
```
Comment (ismael-mendoza) [inline bliss/predict.py:20]: consider merging the 3 models into a model.

File: bliss/report.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+"""Functions to generate metrics on real data."""
+import torch
```
Comment (jeff-regier) [inline bliss/report.py:2]: Would it be OK to include the code from metrics.py and plot.py into report.py (perhaps better called reporting.py), so we can delete those two files, and keep all the reporting code together?

Comment (ismael-mendoza) [reply bliss/report.py:2]: Yes, I like this idea

**Comment by ismael-mendoza** [comment]
Here is a plot of the metrics as a function of magnitude. They get a bit noisy towards the left because the number of objects becomes small, but the overall trend makes sense.

Screen Shot 2021-10-19 at 6 16 00 PM

Just for our reference here is a plot of the true and estimated magnitudes: 

Screen Shot 2021-10-19 at 6 19 54 PM

Screen Shot 2021-10-19 at 6 19 56 PM
